client/src/modules/purchases/modals/status.js

```diff
@@ -6,30 +6,29 @@ PurchaseOrderStatusModalController.$inject = [
 ];
 
 function PurchaseOrderStatusModalController(Instance, Notify, PurchaseOrder, Data) {
-    var vm = this;
+  var vm = this;
 
-    // global variables 
-    vm.purchase = Data;
+  // global variables 
+  vm.purchase = Data;
 
-    vm.status = vm.purchase.is_confirmed ? 'confirmed' : 'not_confirmed';
+  // expose to view 
+  vm.close = Instance.close;
+  vm.submit = submit;
 
-    // expose to view 
-    vm.close = Instance.close;
-    vm.submit = submit;
+  vm.isStored = vm.purchase.status_id === 3 || vm.purchase.status_id === 4;
 
-    // submit the choice 
-    function submit(status) {
-        var confirmed = status === 'confirmed' ? 1 : 0;
-        var data = { is_confirmed: confirmed , is_cancelled: vm.purchase.is_cancelled };
-        updateOrder(vm.purchase.uuid, data);
-    }
+  // submit the choice 
+  function submit(status) {
+    var data = { status_id : vm.status };
+    updateOrder(vm.purchase.uuid, data);
```

Comment:
If you `return updateOrder()` here and `return PurchaseOrder.update()` on line 28, you will get a loading indicator on the modal's submit button.